examples/Cassie/cassie_rbt_state_estimator.cc

```diff
@@ -850,147 +1014,318 @@ void CassieRbtStateEstimator::contactEstimation(
-    output.SetPositionAtIndex(position_index_map_.at("base_y"),
-                              cassie_state->GetPositions()[1]);
-    output.SetPositionAtIndex(position_index_map_.at("base_z"),
-                              cassie_state->GetPositions()[2]);
-    output.SetPositionAtIndex(position_index_map_.at("base_qw"),
-                              cassie_state->GetPositions()[3]);
-    output.SetPositionAtIndex(position_index_map_.at("base_qx"),
-                              cassie_state->GetPositions()[4]);
-    output.SetPositionAtIndex(position_index_map_.at("base_qy"),
-                              cassie_state->GetPositions()[5]);
-    output.SetPositionAtIndex(position_index_map_.at("base_qz"),
-                              cassie_state->GetPositions()[6]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_wx"),
-                              cassie_state->GetVelocities()[0]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_wy"),
-                              cassie_state->GetVelocities()[1]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_wz"),
-                              cassie_state->GetVelocities()[2]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_vx"),
-                              cassie_state->GetVelocities()[3]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_vy"),
-                              cassie_state->GetVelocities()[4]);
-    output.SetVelocityAtIndex(velocity_index_map_.at("base_vz"),
-                              cassie_state->GetVelocities()[5]);
-    // We get 0's cassie_state in the beginning because dispatcher_robot_out is
-    // not triggerred by CASSIE_STATE message.
-    // This wouldn't be an issue when you don't use ground truth state.
-    if (output.GetPositions().head(7).norm() == 0){
-      output.SetPositionAtIndex(position_index_map_.at("base_qw"), 1);
+      // Get kinematics cache for ground truth
+      KinematicsCache<double> cache_gt = tree_.doKinematics(
+          output_gt.GetPositions(), output_gt.GetVelocities(), true);
+
+      imu_pos_wrt_world.head(3) = tree_.transformPoints(
+          cache_gt, imu_pos_, pelvis_idx_, 0);
+      imu_pos_wrt_world.tail(4) = output_gt.GetPositions().segment<4>(3);
+      imu_vel_wrt_world.tail(3) = output_gt.GetVelocities().head(3);
+      imu_vel_wrt_world.head(3) = tree_.transformPointsJacobian(
+          cache_gt, imu_pos_, pelvis_idx_, 0, false)*output_gt.GetVelocities();
+      if (print_info_to_terminal_) {
+        // Debugging print statements
+        cout << "Ground Truth: " << endl;
+        cout << "Positions: " << endl;
+        cout << imu_pos_wrt_world.transpose() << endl;
+        cout << "Orientation (quaternion) : " << endl;
+        cout << output_gt.GetPositions().segment<4>(3).transpose() << endl;
+        cout << "Velocities: " << endl;
+        cout << imu_vel_wrt_world.transpose() << endl;
+      }
+
+      // ofile.open("/home/nanda/DAIR/plotting/ekf.csv",
```

Comment:
This output file logging should not be committed to master

Reply:
Done.